File: Code-Interview-Case/Java/math_test/math.java
class math{
    float result;
    public void add(int a, int b){
        result = a + b;
    }

    public void substract(int a, int b){
        result = a - b;
    }

    public void multiply(int a, int b){
        result = a * b;
    }

    public void divide(int a, int b){
        result = a / b;
    }

    public float get_result(){
        return result;
    }
}
public class hello
{   
    public static void main(String[] args){

        /* Hello World Output */
        System.out.println("Hello World !");

        /* Initialize the object */
        math coba = new math();

        /* Call the addition method*/
        coba.add(1,2);
        coba.get_result();
        System.out.println("Result is : " + coba.result);

        /* Call the substract method*/
        coba.substract(1,2);
        coba.get_result();
        System.out.println("Result is : " + coba.result);

        /* Call the multiply method*/
        coba.multiply(1,2);
        coba.get_result();
        System.out.println("Result is : " + coba.result);

        /* Call the divide method*/
        coba.divide(1,2);
        coba.get_result();
        System.out.println("Result is : " + coba.result);
    }
}

File: Code-Interview-Case/Ruby/math_test/math.rb
# This is the Math Operator Class
class MathOperator
  def add(value1, value2)
    @result = value1 + value2
  end

  def substract(value1, value2)
    @result = value1 - value2
  end

  def divide(value1, value2)
    @result = value1 / value2
  end

  def multiply(value1, value2)
    @result = value1 * value2
  end

  def getresult
    puts @result
  end
end

puts 'Hello World'
puts 'Insert Value'

puts 'Value1 :'
value1 = gets.chomp.to_f

puts 'Value2 :'
value2 = gets.chomp.to_f

c = MathOperator.new

# Addition Method
c.add(value1, value2)
puts 'Addition Result :'
c.getresult

# Substraction Method
c.substract(value1, value2)
puts 'Addition Result :'
c.getresult

# multiply Method
c.multiply(value1, value2)
puts 'Multiply Result :'
c.getresult

# divide Method
c.divide(value1, value2)
puts 'Divide Result :'
c.getresult
